src/scheduler/Analysis.py

```diff
@@ -65,17 +67,37 @@ def add_update_task(self, fo):
             self.check_further_process_or_complete(child)
         self.check_further_process_or_complete(fo)
 
-    def add_task(self, fo):
+    def add_task(self, fo: FileObject):
         '''
         This function should be used to add a new firmware object to the scheduler
         '''
         if fo.scheduled_analysis is None:
             fo.scheduled_analysis = MANDATORY_PLUGINS
         else:
-            shuffle(fo.scheduled_analysis)
-            fo.scheduled_analysis = fo.scheduled_analysis + MANDATORY_PLUGINS
+            fo.scheduled_analysis = self._smart_shuffle(fo.scheduled_analysis + MANDATORY_PLUGINS)
         self.check_further_process_or_complete(fo)
 
+    def _smart_shuffle(self, plugin_list: List[str]) -> List[str]:
+        scheduled_plugins = []
+        remaining_plugins = set(plugin_list)
+        while len(remaining_plugins) > 0:
+            next_plugins = self._get_plugins_without_unmet_dependencies(remaining_plugins, scheduled_plugins)
+            if not next_plugins:
+                logging.error('Error: Could not schedule plugins because dependencies cannot be fulfilled: {}'.format(remaining_plugins))
```

Comment:
the remaining plugins could still run
we need to discuss this with @weidenba

Reply:
If I remember correctly, we have a raise exception switch in the config. We could simply use this and let the user decide.

Reply:
I get the argument, but if I see this correctly, this will only happen when a new plugin is developed. Thus making it a development scenario where I don't think stability is at a premium compared to explicit error handling.